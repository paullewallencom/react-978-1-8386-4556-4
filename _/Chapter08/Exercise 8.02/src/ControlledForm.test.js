import React from 'react';
import {render, fireEvent, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import {ControlledForm} from './ControlledForm';

test('renders correctly', () => {
  const {container} = render(<ControlledForm/>);
  expect(container).toMatchSnapshot();
});

test('loads and displays form', async () => {
  let alert = window.alert
  window.alert = jest.fn()

  //Arrange
  const { container } = render(<ControlledForm />);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');
  const form = container.querySelector('form');

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    })
  });

  await wait(() => {
    fireEvent.submit(form)
  })

  //Assert
  expect(email.value).toBe('example@email.com');
  expect(password.value).toBe('password');

  expect(window.alert).toHaveBeenCalledWith("A name was submitted: example@email.com")

  window.alert = alert
});
