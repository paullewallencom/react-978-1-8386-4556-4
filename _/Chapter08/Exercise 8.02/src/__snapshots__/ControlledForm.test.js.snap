// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    novalidate=""
  >
    <label>
      Name:
      <input
        type="text"
        value=""
      />
    </label>
    <label>
      Password:
      <input
        type="password"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
